### PR TITLE
chore: use `BasicBlockExecutorProvider` instead of `ExecutionStage` in ef-tests

### PR DESCRIPTION
**Pull request by kevaundray**
This PR:

- Replaces ExecutionStage with BasicBlockExecutorProvider
- ~~uses execute_batch to execute all blocks in a batch~~ now uses execute to execute each block sequentially, to test an edge case
- Removes usage of StaticFiles

### CODE REVIEW

File: testing/ef-tests/src/cases/blockchain_test.rs [hunk 1]
```diff
@@ -90,50 +91,47 @@ impl Case for BlockchainTestCase {
...
 
-                // Initialize receipts static file with genesis
-                {
-                    let static_file_provider = provider.static_file_provider();
-                    let mut receipts_writer =
-                        static_file_provider.latest_writer(StaticFileSegment::Receipts).unwrap();
-                    receipts_writer.increment_block(0).unwrap();
-                    receipts_writer.commit_without_sync_all().unwrap();
-                }
-
```
Comment (kevaundray) [inline testing/ef-tests/src/cases/blockchain_test.rs:112]: If staticfile does make a difference, it might make sense to initialize the transactions static file too for completeness

**Comment by kevaundray** [comment]
ef-tests (state tests) seem to be about 20-30 seconds faster -- though I only took a sample size of 3 random PRs

File: testing/ef-tests/src/cases/blockchain_test.rs (hunk 1 above)
Comment (kevaundray) [reply testing/ef-tests/src/cases/blockchain_test.rs:112]: Seems this PR is still faster -- https://github.com/paradigmxyz/reth/pull/15542#issuecomment-2779768262

File: testing/ef-tests/src/cases/blockchain_test.rs [hunk 2]
```diff
@@ -90,50 +91,44 @@ impl Case for BlockchainTestCase {
...
-                    ExecInput { target: last_block.as_ref().map(|b| b.number), checkpoint: None },
-                );
+                    provider.insert_block(recovered_block.clone(), StorageLocation::Database)?;
+
+                    blocks.push(recovered_block);
+                }
+                let last_block = blocks.last().cloned();
+
+                // Initialize executor with state
+                let basic_block_executor = EthExecutorProvider::ethereum(chain_spec);
```
Comment (kevaundray) [inline testing/ef-tests/src/cases/blockchain_test.rs:120]: This should be more accurately named executor_provider, since its not an executor

File: testing/ef-tests/src/cases/blockchain_test.rs [hunk 3]
```diff
@@ -90,50 +91,47 @@ impl Case for BlockchainTestCase {
...
+                    let executor = executor_provider.executor(state_db);
+
+                    // Execute all blocks in a batch
+                    if let Ok(output) = executor.execute(&block) {
+                        provider.write_state(
+                            &ExecutionOutcome::single(block.number, output),
+                            OriginalValuesKnown::Yes,
+                            StorageLocation::Database,
+                        )?;
+                    }
```
Comment (kevaundray) [inline testing/ef-tests/src/cases/blockchain_test.rs:133]: @mattsse / @rkrasiuk Any insights as to why the following doesn't work?

Changing `let state_provider = LatestStateProviderRef::new(&provider)` makes the tests pass, however I am trying to execute each block using the same code pattern as debug_executionWitness; ie get the state of the chain when the parent block was executed and execute the child block using that state.
